src/js/git/commands.js

```diff
@@ -568,7 +569,10 @@ var commandConfig = {
 
```

Comment:
so the command parsing is a little weird. itll grab the first argument *after* the `--no-ff` and put it into this options map.

but for boolean flags that doesnt make sense, so we tack them on manually. so i would do

command.getGeneralArgs().concat(commandOptions['--no-ff'] || []).concat(commandOptions['--squash'] || [])

and then it might work